Source/CsDebugScript.Engine/Engine/IBreakpoint.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+
+namespace CsDebugScript.Engine
+{
+    /// <summary>
+    /// Indicates state of the debugger after this breakpoint is hit.
+    /// </summary>
+    public enum BreakpointEventStatus
+    {
+        /// <summary>
+        /// Continues execution after executing action assosiated to this breakpoint.
+        /// </summary>
+        ReleaseDebugger,
+        /// <summary>
+        /// Breaks execution until user again issues continue debugger action.
+        /// </summary>
+        BreakDebugger,
+    }
+
+    /// <summary>
+    /// Breakpoint can't be defined with this expression.
+    /// </summary>
+    public class InvalidBreakpointLocation : Exception
+    {
+        /// <summary>
+        /// Passing message to base exception.
+        /// </summary>
+        /// <param name="message"></param>
+        public InvalidBreakpointLocation(string message) : base(message)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Interface for breakpoints.
+    /// </summary>
+    public interface IBreakpoint
+    {
+        /// <summary>
+        /// Removes this breakpoint.
+        /// </summary>
+        void Remove();
```

Comment:
I'm not sure if it is OK that Breakpoint knows how to remove itself. Maybe it is better to have function in Debugger class that would remove breakpoint.